fhir-api/authentication.py

```
@@ -1,11 +1,23 @@
 from flask import request
 from functools import wraps
-import jwt
 import os
+import requests
 
 from errors import AuthenticationError
 
-JWT_PUBLIC_KEY = os.getenv("JWT_PUBLIC_KEY").replace("\\n", "\n")
+TOKEN_INTROSPECTION_URL = os.getenv("TOKEN_INTROSPECTION_URL")
+
+
+def validate_token(token, scope=None):
+    payload = {"token": token}
+    if scope is not None:
+        payload["scope"] = scope
+
+    response = requests.post(TOKEN_INTROSPECTION_URL, data=payload)
+
+    validation = response.json()
```

maybe we can throw an error using the error message from hydra to give more feedback if the token is invalid ?

---

Hydra doesn't give an error but a simple response body with a few fields and when the token is not active, I think it just give {active: false}